src/bindings/jna/libelektra4j/src/main/java/org/libelektra/exception/KDBException.java

```diff
@@ -0,0 +1,90 @@
+package org.libelektra.exception;
+
+import org.libelektra.Key;
+import org.libelektra.exception.model.WarningEntry;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * Custom KDB exception class being used for I/O errors
+ */
+public abstract class KDBException extends java.io.IOException {
+
+	private static final long serialVersionUID = 1L;
+
+	final transient Key errorKey;
+	private Collection<WarningEntry> warnings;
+
+	public KDBException(final Key k) {
+		errorKey = k;
+		Key warningsKey = k.getMeta("warnings");
+		warnings = new ArrayList<>();
+		if (warningsKey.isNull()) {
+			return;
+		}
+		final int nr = warningsKey.getInteger();
+		for (int i = 0; i <= nr; i++) {
+			warnings.add(new WarningEntry(k, nr));
+		}
+	}
+
+	public Key getErrorKey() {
+		return errorKey;
+	}
+
+	public String getErrorCode() {
```

Comment:
There is a huge mess with code/number. Please use always the same.

Reply:
What do you mean with "mess"? Do I sometimes write "code" and sometimes "number"?

Reply:
Yes, see the line below.